### ISSUE TITLE
Add some parameterless contructors to some CandlestickPatterns 

### ISSUE THREAD

**Issue by Jay-Jay-D** [QuantConnect/Lean#563]
I found some problems using Reflection to instantiate some CandleStickPatterns, specifically the AbandonedBaby, DarkCloudCover, EveningDojiStar, EveningStar, MatHold, MorningDojiStar and MorningStar.

When I try this:
```cs
dynamic indicator = Activator.CreateInstance(pattern);
```
whith the patterns I mention above, I receive the `No parameterless constructor defined for this object.` error.

In principle there is a constructor that can be used without parameters, e.g.:
```cs
public AbandonedBaby(decimal penetration = 0.3m)
            : this("ABANDONEDBABY", penetration)
```
But it seems that Reflection doesn't take the default value constructor as parameterless.

The solution is simply adding a parameterless constructor to the CandleStickPatterns like this:
```cs
public AbandonedBaby()
            : this("ABANDONEDBABY")
```
So, what do you think? Is worthy to make a PR with the parameterless constructors?

**Comment by mchandschuh** [comment]
Fyi, using reflection you could get the default values and invoke the constructor with those values. 

That being said, defining a default ctor that simply calls into another with all optional arguments seems perfectly  fine to me.

**Comment by mchandschuh** [comment]
Another note, this may cause ambiguity warnings for users who explicitly call the default ctor, something to the extent of overload hidden by method with optional parameter, which in this case, is perfectly fine since the 'hiding' method calls the hidden 'method'  with all defaults.

**Comment by Jay-Jay-D** [comment]
Maybe the solution can be like this:
```cs
public AbandonedBaby(string name, decimal penetration = 0.3m) 
    : base(...)
{ }

public AbandonedBaby(decimal penetration)
    : this("ABANDONEDBABY", penetration)
{ }

public AbandonedBaby()
    : this("ABANDONEDBABY")
{ }
```

**Comment by mchandschuh** [comment]
I believe that pattern will side step the warning message I mentioned.

**Comment by Jay-Jay-D** [comment]
Yes, that's the idea. 

If you agree I can make the PR with those changes for the indicators I mention above.

**Comment by jaredbroad** [comment]
Thank you, please proceed with the PR @Jay-Jay-D

**Comment by Jay-Jay-D** [comment]
#569 Done.

**Comment by jaredbroad** [comment]
Thank you JJ, the last commit failed TravisCI with message:

QCAlgorithm.Indicators.cs(569,101): error CS1525: Unexpected symbol
`selector'

**Comment by Jay-Jay-D** [comment]
Jared, I'm so sorry, today was one of those days...

The problem was because I left unfinished the fix to issue #565, now wjth the latest commit it [compiles](https://travis-ci.org/QuantConnect/Lean/jobs/162448700).

Sorry for the mess.
